app/controllers/assessment360_controller.rb

```diff
@@ -59,21 +57,27 @@ def all_students_all_reviews
                                                    @meta_review_info_per_stu)
       end
       # calculate average grade for each student on all assignments in this course
-      if @teammate_review_info_per_stu[1] > 0
-        temp_avg_grade = @teammate_review_info_per_stu[0] * 1.0 / @teammate_review_info_per_stu[1]
-        @teammate_review[cp.id][:avg_grade_for_assgt] = temp_avg_grade.round.to_s + '%'
-      end
-      if @meta_review_info_per_stu[1] > 0
-        temp_avg_grade = @meta_review_info_per_stu[0] * 1.0 / @meta_review_info_per_stu[1]
-        @meta_review[cp.id][:avg_grade_for_assgt] = temp_avg_grade.round.to_s + '%'
-      end
+      review_info_per_student(cp,@teammate_review_info_per_stu, @teammate_review)
+      review_info_per_student(cp,@meta_review_info_per_stu, @meta_review)
     end
     # avoid divide by zero error
-    @assignments.each do |assignment|
-      temp_count = @overall_teammate_review_count[assignment.id]
-      @overall_teammate_review_count[assignment.id] = 1 if temp_count.nil? or temp_count.zero?
-      temp_count = @overall_meta_review_count[assignment.id]
-      @overall_meta_review_count[assignment.id] = 1 if temp_count.nil? or temp_count.zero?
+    avoid_divide_by_zero_error(@assignments,@overall_teammate_review_count,@overall_meta_review_count)
+  end
+
+  def avoid_divide_by_zero_error (assignments,overall_teammate_review_count,overall_meta_review_count)  
```

Comment:
This is not a good name for this method.  I cannot understand what it is actually doing.  Probably would be better to break it up into a method that could operate on either teammate_review_count or meta_review_count, and would be called once for each.  
    Even better: instead of teammate_review_count (or overall_teammate_review_count), have a teammate_reviews object and use teammate_reviews.count to return the count.  But that may be too much to ask, since there's only one day left.

Reply:
The hash value for a student is calculated based on the teammate_review_count. Having just one function that can be called with teammate or meta variables might want us to do an extra check if the value passed is a teammate variable or a meta variable